lib/checks/shared/heading-whitespace-only.js

```diff
@@ -0,0 +1,24 @@
+const text = axe.commons.text.accessibleTextVirtual(virtualNode, {
+	sanitize: false
+});
+const hasBreakingSpaceElement =
+	axe.utils.querySelectorAll(virtualNode, 'br, wbr').length > 0;
+
+if (text.length === 0 && !hasBreakingSpaceElement) {
+	return true;
+}
+
+const spaceSeparatorCharCodes = [
+	160, // non-breaking space
+	8194, // en space
+	8195, // em space
+	8201 // thin space
+];
+
+const hasSpaceSeparator =
+	text
+		.split('')
+		.map(char => char.charCodeAt(0))
+		.filter(charCode => spaceSeparatorCharCodes.includes(charCode)).length > 0;
```

Comment:
This looks wrong to me. `<h3>Foo&160;bar</h3>` shouldn't fail this rule. I think what this should be is this:

```js
const textCharCodes = text.split('').map(char => char.charCodeAt(0))
const hasOnlySpaceSeparator = textCharCodes.every(charCode => {
  return spaceSeparatorCharCodes.includes(charCode)
})
```

Reply:
Should `<h3>   &160;   </h3>` also fail the rule?